packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts

```diff
@@ -58,17 +59,33 @@ async function createDataObject<TObj extends PureDataObject, I extends DataObjec
     let runtimeClass = runtimeClassArg;
 
     // request mixin in
+    // runtimeClass = mixinRequestHandler(
+    //     async (request: IRequest, runtimeArg: FluidDataStoreRuntime) => {
+    //         const maybeRouter: FluidObject<IFluidRouter> | undefined = await runtimeArg.entryPoint?.get();
+    //         assert(maybeRouter !== undefined, "entryPoint should have been initialized by now");
+    //         assert(maybeRouter?.IFluidRouter !== undefined, "Data store runtime entryPoint is not an IFluidRouter");
+    //         return maybeRouter?.IFluidRouter.request(request);
+    //     },
+    //     runtimeClass);
     runtimeClass = mixinRequestHandler(
         async (request: IRequest, runtimeArg: FluidDataStoreRuntime) =>
             (await PureDataObject.getDataObject(runtimeArg)).request(request),
             runtimeClass);
 
```

Comment:
Sticking to `PureDataObject.getDataObject()` in the request handler to have that be the place where we call `finishInitialization()` on the data object. Accessing the entrypoint in the data store runtime still returns the data object instance, but it doesn't guarantee that it will be fully initialized. In the new-container flow I think we still get a fully initialized data object no matter what because of lines 110-112 in this file, but in the load-existing-container flow, I don't see a good way of guaranteeing that you'd get a handle to a fully initialized object unless the guidance is "do not grab the handle directly from the DS entrypoint for a DataObject, go through `PureDataObject.getDataObject()` instead". Which seems a bit at odds with the objective of making the entrypoint *the* way of getting to the user code tied to the data store.

Reply:
Outdated